url-updater/src/main/java/com/devonfw/tools/ide/url/updater/tomcat/TomcatUrlUpdater.java

```diff
@@ -0,0 +1,37 @@
+package com.devonfw.tools.ide.url.updater.tomcat;
+
+import java.util.regex.Pattern;
+
+import com.devonfw.tools.ide.common.OperatingSystem;
+import com.devonfw.tools.ide.url.model.folder.UrlVersion;
+import com.devonfw.tools.ide.url.updater.WebsiteUrlUpdater;
+
+/**
+ * {@link WebsiteUrlUpdater} for Tomcat.
+ */
+public class TomcatUrlUpdater extends WebsiteUrlUpdater {
+  @Override
+  protected String getTool() {
+
+    return "tomcat";
+  }
+
+  @Override
+  protected void addVersion(UrlVersion urlVersion) {
+
+    doAddVersion(urlVersion, "https://archive.apache.org/dist/tomcat/tomcat-${major}/v${version}/src/apache-tomcat-${version}-src.${ext}");
+  }
+
+  @Override
+  protected String getVersionUrl() {
+
+    return "https://github.com/apache/tomcat/tags";
+  }
+
+  @Override
+  protected Pattern getVersionPattern() {
+
+    return Pattern.compile("[0-9]+\\.[0-9]+\\.[0-9]+(\\.|-)M[0-9]+");
```

Comment:
`M` stands for a milestone release that is rather in-official. It should not be a mandatory part of the version pattern.

Reply:
Great. Thanks. Resolved.